src/http-client.ts

```diff
@@ -54,11 +54,21 @@ export const initHttpClient = ({
       const parseAsJson =
         response.headers.get("content-type") === "application/json";
 
-      const data = await (parseAsJson ? response.json() : response.text());
```

Comment:
I agree that this feels wrong, and the API endpoints are wrong here for responding with `content-type: application/json` when the content isn't actually json.

What do you think about adding this check here instead of swallowing JSON errors?

```js
const parseAsJson =
  response.headers.get("content-length") !== "0" &&
  response.headers.get("content-type") === "application/json";
```

Reply:
Can we verify the response headers?